File: jvm/core/main/scala/com/webank/eggroll/core/deepspeed/store/meta/RendezvousStoreDestroyRequest.scala
package com.webank.eggroll.core.deepspeed.store.meta

import com.google.protobuf.ByteString
import com.webank.eggroll.core.meta.Deepspeed

import scala.language.implicitConversions

case class RendezvousStoreDestroyRequest(prefix: String)

object RendezvousStoreDestroyRequest {
  implicit def serialize(src: RendezvousStoreDestroyRequest): Array[Byte] = {
    val builder = Deepspeed.StoreDestroyRequest.newBuilder()
      .setPrefix(src.prefix)
    builder.build().toByteArray
  }

  implicit def deserialize(byteString: ByteString): RendezvousStoreDestroyRequest = {
    val src = Deepspeed.StoreDestroyRequest.parseFrom(byteString)
    RendezvousStoreDestroyRequest(
      prefix = src.getPrefix
    )
  }
}

File: jvm/new_core/src/main/java/com/eggroll/core/deepspeed/store/RendezvousStoreDestroyRequest.java
package com.eggroll.core.deepspeed.store;

import com.eggroll.core.pojo.RpcMessage;
import com.google.protobuf.InvalidProtocolBufferException;
import com.webank.eggroll.core.meta.Deepspeed;
import lombok.Data;

@Data
public class RendezvousStoreDestroyRequest implements RpcMessage {
    private String prefix;

    @Override
    public byte[] serialize() {
        Deepspeed.StoreDestroyRequest.Builder builder = Deepspeed.StoreDestroyRequest.newBuilder()
                .setPrefix(this.getPrefix());
        return builder.build().toByteArray();
    }

    @Override
    public void deserialize(byte[] data){
        try {
            Deepspeed.StoreDestroyRequest src = Deepspeed.StoreDestroyRequest.parseFrom(data);
            this.prefix = src.getPrefix();
        } catch (InvalidProtocolBufferException e) {
            e.printStackTrace();
        }
    }
}